src/components/tutorials-sidebar/index.tsx

```diff
@@ -0,0 +1,89 @@
+import { Fragment } from 'react'
+import VisuallyHidden from '@reach/visually-hidden'
+import { IconChevronLeft16 } from '@hashicorp/flight-icons/svg-react/chevron-left-16'
+import isAbsoluteUrl from 'lib/is-absolute-url'
+import StandaloneLink from 'components/standalone-link'
+import SkipToMainContent from 'components/sidebar/components/sidebar-nav/skip-to-main-content'
+import { SidebarNavLink } from 'components/sidebar/components/sidebar-nav/sidebar-nav-menu-item'
+import {
+  ListItemProps,
+  SectionListProps,
+  SectionTitleProps,
+  TutorialSidebarProps,
+} from './types'
+import s from './tutorials-sidebar.module.css'
+import { HorizontalRule as DocsHorizontalRule } from 'components/sidebar/components/sidebar-nav/sidebar-nav-menu-item'
+
+function TutorialsSidebar({
+  backToLink,
+  children,
+  ariaLabelledBy,
+}: TutorialSidebarProps) {
+  return (
+    <nav className={s.root} aria-labelledby={ariaLabelledBy}>
+      <div className={s.backToLink}>
+        <StandaloneLink
+          href={backToLink.href}
+          text={backToLink.text}
+          icon={<IconChevronLeft16 />}
+          iconPosition="leading"
+          textSize={200}
+        />
+      </div>
+
+      <div className={s.itemsContainer}>
+        <SkipToMainContent />
+        {children}
+      </div>
+    </nav>
+  )
+}
+
+function SectionList({ items }: SectionListProps) {
+  return (
+    <ul className={s.listRoot}>
+      {items.map(({ text, href, isActive, isExternal }: ListItemProps) => {
+        return (
+          <ListItem
+            key={`${text}${href}`}
+            text={text}
+            href={href}
+            isActive={isActive}
+            isExternal={isExternal}
+          />
+        )
+      })}
+    </ul>
+  )
+}
+
+function ListItem({ href, isActive, text, isExternal }: ListItemProps) {
+  const parsedIsExternal =
+    typeof isExternal == 'boolean' ? isExternal : isAbsoluteUrl(href)
+  const hrefOrFullPath = parsedIsExternal ? { href } : { fullPath: href }
+  return <SidebarNavLink item={{ isActive, title: text, ...hrefOrFullPath }} />
+}
```

Comment:
Curious on why `isExternal` is being added rather than using the two separate props (`href` & `fullPath`) that the docs sidebar currently relies on, and are in effect being used here anyways?

Reply:
Great question. Here's the rationale:
1. I wanted to modify `SidebarNavLink` as little as possible, so leave the `href` & `fullPath` in place...
2. This meant that to get an external icon, I'd have to pass `fullPath`; to omit it I'd have to pass `href`...
3. I wanted the props for `ListItem` to be as simple as possible - `href` for all links. We can handle this pretty nicely with `MaybeInternalLink` so figured this was worth it...
5. Initially I had `isExternal` not exposed as a prop, and directly determined it from `isAbsoluteUrl`. Worked fine, mostly...
6. But then realized there will be some non-internal links marked with the external link icon (namely `All Tutorials`) which need to be marked with the external icon...
7. So that led me to expose `isExternal` as an optional prop, with the default behaviour being show it only if `isAbsoluteUrl(href)`

I considered that we could, instead of handling this at the `ListItem` component level, lift up the responsibility to where `ListItem` is used, and ultimately to whatever data massaging we do in the tutorials view. However I think the data massaging would be more work than what we're doing here - we'd have to either author content with separate `href` and `fullPath` properties depending on link location, or traverse the incoming data or something.

Let me know if that makes sense 🙇‍♂️

Reply:
Thank you for the thorough context! Makes sense. 🙏🏻

> It's possible that we could eliminate the need for the isExternal prop by being super consistent in design on only showing the "external" link for links that are outside dev-portal.

Yep, this is what we need to do. "All Tutorials" might be "external" to the current page, but it is not external to the whole site. The external link icon should only be used for links that are totally outside of the DevDot site.

You don't need to make any changes in this PR, can save it for a follow-up if you've got one coming. 😊

Reply:
👍 Sounds good, I think it makes sense to follow up on this.

This feels like something we should possibly raise with design - it's definitely possible this was an unintentional slip in the sidebar mockup, though I think it's also possible the design intent here doesn't always match the model of external link icons being strictly for separate domains, and may align to a fuzzier concept of using external link icons for separate experiences.